### ISSUE TITLE
UI Application Details Tree not showing the complete name of applications/resources

### ISSUE THREAD

**Issue by prsrus216** [argoproj/argo-cd#19110]
# Summary

Argo UI Application Details Tree page not showing the Application/Resource Full Names . Its only showing the truncated names and had to hover the mouse to see the complete name or click on "Truncate Resources name left" option to see last portion of the name. Provided screenshot with the resource name highlighted which is not showing the complete name.

<img width="1082" alt="Screenshot 2024-07-18 at 9 34 28 AM" src="https://github.com/user-attachments/assets/00491838-c3c3-44ed-ae63-a7e1af1dda26">

Hope can this is customized to have names showing as wrap text to show the complete name. Probably may need to adjust the CSS and add the custom CSS to configmap.

# Motivation

Its hard to hover the mouse to each resource to see the complete name and hence if the names are showing as wrap its easy to read through. 

# Proposal

Hope can this is customized to have names showing as wrap text to show the complete name. Probably may need to adjust the CSS and add the custom CSS to configmap. Appreciated if anyone can help on how this can be added.

**Comment by christianh814** [comment]
What is your reason for needing to see the whole name on the overview? As you said you can hover over it to see the full name.

**Comment by prsrus216** [comment]
Hi @christianh814 Thanks You for the response. We have most of the app/resource names prefix is same but the middle and last part of the name is different. Hence, if we need to select the required app then need to hover the mouse to the complete list one at a time to find the correct app/resource and its way difficult when the list has many. Thought of can have full name displaced and can just scroll through to read the name and select the correct app.

**Comment by prsrus216** [comment]
@keithchong  Thank You for the response. I have tried left/right justification but still not able to get the required name viewable as we are missing some middle name when it has lengthy app name. Probably adding another toggle option to see full name might helps or adjust some CSS for the default view.

**Comment by prsrus216** [comment]
Hi @keithchong  I just have tried adding the below config to argocd-cm configmap but NO luck.

resource.customizations: |
              applications:
                css: |
                  .application-resource-tree {
                    position: relative;
                    overflow: visible;
                  }
                  .__node-title {
                    white-space: normal;   
                    overflow: visible;    
                    text-overflow: clip;   
                  }

**Comment by keithchong** [comment]
I think code changes might have to be done.  Basically, that option to truncate left or right was added so users can readily see the names in the view (typically, the trailing part was the differentiator, at least from what I recall of the original issue). However, you're saying this feature is still insufficient for your needs.  So, we probably need a different design/solution.

My two cents: I'm not in favour of showing the entire name, wrapped within the node 'border', because the names can be long. I don't like the nodes to be of different heights.

**Comment by prsrus216** [comment]
@keithchong Yes, hovering over each service to see the full name is tedious, especially when there are many in the list. Could we implement a feature flag that allows users to enable or disable the full name/wrap view? This would let users enable text wrap to see the full name and disable it to maintain uniform node heights with left/right justification.

**Comment by ba32107** [comment]
I managed to make this work by following this guide:
https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/custom-styles.md

Here's the CSS:
```CSS
.application-resource-tree__node {
  height: unset !important;
}
.application-resource-tree__node-content {
  white-space: normal;
}
```

Result:
![image](https://github.com/user-attachments/assets/99df6e01-4b3c-453d-9f44-097c839264ed)

**Comment by ba32107** [comment]
+1 for adding a setting to toggle this behavior. We have Argo apps with ~30+ resources and it's really difficult to find the ones I'm looking for, as they all have reasonably long names

**Comment by surajyadav1108** [comment]
> +1 for adding a setting to toggle this behavior. We have Argo apps with ~30+ resources and it's really difficult to find the ones I'm looking for, as they all have reasonably long names

I will add a toggle Similar to truncate in tool panel

**Comment by ba32107** [comment]
Hi @surajyadav1108 , thanks for raising that PR. Just FYI, I also implemented custom CSS to show a long application name (in the main view that shows the application tiles):

![image](https://github.com/user-attachments/assets/e33be033-0b7e-41ee-a1c5-2340cd3b8dbc)

```CSS
.columns:has(> .argo-icon-git), .columns:has(> .argo-icon-helm) {
  display: flex;
  align-items: center;
}
.columns > .applications-list__title {
  white-space: normal !important;
}
```
Perhaps you can add this to the PR?

**Comment by surajyadav1108** [comment]
@ba32107 


I have added this in the PR but only having this 
 white-space: normal !important;
}
was not working for me so i have the css as this which gives the same kind of result


  ```
  .applications-list__title {
        white-space: normal !important;
        word-break: break-word; 
        overflow-wrap: break-word; 
        hyphens: auto; 
        line-height: 1.4; 
        width: 100%; 
    }
```

**Comment by prsrus216** [comment]
I have added below custom styles to the config and its working fine.

styles: |
            .application-resource-tree__node {
              height: unset !important;
            }
            .application-resource-tree__node-content {
              white-space: normal;
            }
However, some blocks are overlapping when the app name is too big like below. Any suggestions please?

![Image](https://github.com/user-attachments/assets/05efb55e-24be-41f2-9a97-abf49ccb56b4)

**Comment by ba32107** [comment]
I think I added some more CSS to the little labels so they're always on top. I just accepted that there are no gaps between the boxes.

You could also decrease the font size, that should help with the box sizing.

**Comment by pkazi** [comment]
How to default set to show full names always ?
